Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.h

```diff
@@ -129,11 +129,14 @@ namespace AZ::Dom
         //! @return The DomPrefixTree that is detached.
         DomPrefixTree<T> DetachSubTree(const Path& path);
 
-        //! Attaches a subtree provided at the node that matches the provided path. Attaching will overwrite the node at the path.
-        //! @param path The path which corresponds to the node at which the subtree should be attached.
-        //! @param subTree The subtree to attach at the provided path.
-        //! @return True if the subTree was attached successfully. Return false otherwise.
-        bool AttachSubTree(const Path& path, DomPrefixTree&& subTree);
+        //! Overwrites the subtree at the path and replaces it with the given subtree.
+        //! Old node and its children at the path would be destroyed. The new subtree will be moved to the path.
+        //! @param path The path which corresponds to the root node to which the subtree would be overwritten.
+        //! @param subtree The subtree to move to the provided path.
+        //! @param createParentsIfNeeded Create parent entries for the provided path if the flag is on.
+        //! @return True if the subtree was overwritten successfully, false if createParentsIfNeeded is off and no target path exists.
+        bool OverwritePath(const Path& path, DomPrefixTree&& subtree, bool createParentsIfNeeded = false);
```

Comment:
nit : can make the bool name compact by calling it 'shouldCreateParents' or 'createParents'